### PR TITLE
Use aciton.yaml and omit default runner

### PR DESCRIPTION
**Pull request by KevDevSha**
Use Action.yamls which will allows us to omit the runs-on attribute so we can use custom runner sizes

tested with https://github.com/databricks-mosaic/runtime-private-plugins/pull/50/checks

### CODE REVIEW

**Comment by b-chu** [comment]
These files should get deleted?
https://github.com/mosaicml/ci-testing/blob/main/.github/workflows/pytest-cpu.yaml

File: .github/actions/pytest-cpu/action.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  code_eval_apikey:
+    required: false
+  gcs_key:
+    required: false
+  gcs_secret:
+    required: false
+  azure_account_name:
+    required: false
+  azure_account_access_key:
+    required: false
```
Comment (b-chu) [inline .github/actions/pytest-cpu/action.yaml:51]: Do these need to be secrets?

**Review by b-chu — COMMENTED** [review]
Thanks for the PR! Include a manual test from another repo using this action?

File: .github/actions/pytest-cpu/action.yaml (hunk 1 above)
Comment (KevDevSha) [reply .github/actions/pytest-cpu/action.yaml:51]: The `composite` workflow would not allow secrets so I had to switch them to inputs. Although, anything that is sent via `${{ secrets.x }}` will be masked.

**Comment by KevDevSha** [comment]
> These files should get deleted? https://github.com/mosaicml/ci-testing/blob/main/.github/workflows/pytest-cpu.yaml

These are to be used by the public actions, the new actions/pr-gpu/cpu test is to be used by the private actions. Once this is all fleshed out, we can migrate the public actions over next release 👍

File: .github/actions/pytest-cpu/action.yaml (hunk 1 above)
Comment (KevDevSha) [reply .github/actions/pytest-cpu/action.yaml:51]: image

**Comment by KevDevSha** [comment]
> Thanks for the PR! Include a manual test from another repo using this action?

@b-chu https://github.com/databricks-mosaic/runtime-private-plugins/pull/50/checks

**Comment by b-chu** [comment]
> Once this is all fleshed out, we can migrate the public actions over next release
Can public actions use the same thing as the private actions? Seems like we can delete the workflows if everything can use actions instead

File: .github/workflows/pytest-gpu.yaml [hunk 2]
```diff
@@ -12,6 +12,10 @@ on:
       git_repo:
         required: true
         type: string
+      ci_repo_ref:
+        required: false
```
Comment (b-chu) [inline .github/workflows/pytest-gpu.yaml:16]: Make this required. It seems easy for calling workflows to forget to set this and end up with unintended behavior

**Review by b-chu — COMMENTED** [review]
Some more comments :)

**Comment by KevDevSha** [comment]
@b-chu deleted!

**Review by b-chu — APPROVED** [review]
Thanks for the PR!

**Pull request merged**